src/Microsoft.Graph.Cli.Core/Authentication/DeviceCodeLoginService.cs

```diff
@@ -0,0 +1,20 @@
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Graph.Cli.Core.IO;
+using Microsoft.Graph.Cli.Core.Utils;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Graph.Cli.Core.Authentication;
+
+public class DeviceCodeLoginService : LoginServiceBase {
+    private DeviceCodeCredential credential;
+
+    public DeviceCodeLoginService(DeviceCodeCredential credential, IPathUtility pathUtility) : base(pathUtility) {
+        this.credential = credential;
+    }
+
+    protected override async Task<AuthenticationRecord> DoLoginAsync(string[] scopes) {
+        return await credential.AuthenticateAsync(new TokenRequestContext(scopes));
```

Comment:
this is probably not specific to the device code flow and could be reused for other flows, but we can always rename as we add more flows.

Reply:
Yeah. We can move it into the login base abstract class if needed